Source/WebCore/dom/FullscreenManager.cpp

```diff
@@ -503,7 +504,10 @@ bool FullscreenManager::willEnterFullscreen(Element& element, HTMLMediaElementEn
 
     ASSERT(&element == m_pendingFullscreenElement);
     m_pendingFullscreenElement = nullptr;
+
     m_fullscreenElement = &element;
+    updatePageFullscreenStatusIfTopDocument();
+    protectedPage->setTopDocumentHasFullscreenElement(true);
```

Comment:
I still think this line needs to be removed.  It was replaced by the previous line which is more correct.